core/src/avm1/globals.rs

```diff
@@ -69,3 +112,143 @@ pub fn create_globals<'gc>(gc_context: MutationContext<'gc, '_>) -> Object<'gc>
 
     globals
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::avm1::activation::Activation;
+    use crate::avm1::Error;
+    use crate::backend::audio::NullAudioBackend;
+    use crate::backend::navigator::NullNavigatorBackend;
+    use crate::display_object::DisplayObject;
+    use crate::movie_clip::MovieClip;
+    use gc_arena::rootless_arena;
+    use rand::{rngs::SmallRng, SeedableRng};
+
+    macro_rules! test_std {
+        ( $test: ident, $fun: expr, $version: expr, $($args: expr => $out: expr),* ) => {
+            #[test]
+            fn $test() -> Result<(), Error> {
+                with_avm($version, |avm, context, this| {
+
+                    $(
+                        assert_eq!($fun(avm, context, this, $args), $out);
+                    )*
+
+                    Ok(())
+                })
+            }
+        };
+    }
```

Comment:
Is this macro the same as the one in `math.rs`? Can we move it to a util module and use it in both?

Reply:
More or less. I think I originally did a copy and paste as I wasn't sure how much was going to change, but if I recall correctly all it ended up being was returning an extra parameter. So yes, no reason we can't share these :+1:

Reply:
The macro is actually slightly different. I've moved with_avm to a shared module though. The difference between the 2 macros is that the one in `math.rs` creates a math object and then looks up the function by function name, where as this one has a function pointer passed in.